src/components/Checkout/Shipping.tsx

```diff
@@ -0,0 +1,226 @@
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import { useCallback, useContext, useEffect, useRef, useState } from 'react';
+import * as Yup from 'yup';
+
+import ShoppingCart from '../../assets/svg/shoppingCart.svg';
+
+import { FORM } from '../../constants/form';
+import { TRANSLATE } from '../../constants/languages';
+import { cartContext } from '../../context/cartContext';
+import { ICartContext } from '../../context/Types';
+import Form from '../Form/Form';
+import OrderSummaryListItem from './OrderSummaryListItem';
+import { IShippingProps } from './Types';
+
+const Shipping: React.FC<IShippingProps> = ({ promocodes }) => {
+  const { locale } = useRouter();
+
+  const isServer = typeof window === 'undefined';
+
+  const [orderSummaryListHeight, setOrderSummaryListHeight] = useState('0');
+
+  const orderListRef = useRef<HTMLDivElement>(null);
+
+  const { cart } = useContext(cartContext) as ICartContext;
+
+  const toggleOrderList = (mode: boolean) => {
+    setOrderSummaryListHeight(
+      mode ? `${orderListRef.current?.scrollHeight}px` : '0'
+    );
+  };
+
+  const checkDiscountCode = (enteredCode: string) => {
+    const activeCodes = promocodes.filter((item) => item.code === enteredCode);
+    if (activeCodes.length) {
+      const activeCode = activeCodes[0];
+      setTotalCheckout(
+        activeCode.inPercent
+          ? calcTotalCheckout() *
+              (1 - Number.parseFloat(activeCode.discount) / 100)
+          : calcTotalCheckout() - Number.parseFloat(activeCode.discount)
+      );
+      console.log(activeCodes.length);
+    } else {
+      setTotalCheckout(calcTotalCheckout());
+    }
+  };
+
+  const calcTotalCheckout = useCallback(
+    () =>
+      cart.reduce(
+        (counter, cartItem) =>
+          counter + (cartItem.amount * Number.parseFloat(cartItem.price) || 0),
+        0
+      ),
+    [cart]
+  );
+
+  const [totalCheckout, setTotalCheckout] = useState(calcTotalCheckout());
+
+  useEffect(() => {
+    const handleResize = () => {
+      setOrderSummaryListHeight(window.innerWidth < 1024 ? '0' : '100%');
```

Comment:
use css for viewport resize, using resize eventListener can have a bad effect on ui